runtime/jaeger_metrics.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zanzibar
+
+import (
+	"sync"
+
+	"github.com/uber-go/tally"
+	jaeger_metrics "github.com/uber/jaeger-lib/metrics"
+)
+
+type jaegerGauge struct {
+	gauge tally.Gauge
+}
+
+func (j *jaegerGauge) Update(value int64) {
```

Comment:
Good to know. It seems that jaeger-lib creates a wrapper struct each time we call Counter/Gauge/Timer, and those wrapper structs are not cached as they are in tally. The only thing that really needs a wrapper struct is Gauge as the interface is different, and I am caching it here. But I agree that we don't need reimplement this and those wrapper structs are probably not a big deal. Will reuse jaeger-lib.